File: minimum_common/min.py
from typing import List


class Solution:
    def getCommon(self, nums1: List[int], nums2: List[int]) -> int:
        common_list: List[int] = []

        for num in nums1:
            if binary_search(nums2, num):
                common_list.append(num)

        if len(common_list) == 0:
            return -1
        else:
            return min(common_list)


def binary_search(nums: List[int], target: int) -> bool:
    i, j = 0, len(nums) - 1

    while i <= j:
        mid = i + (j - i) // 2

        if nums[mid] < target:
            i = mid + 1
        elif nums[mid] > target:
            j = mid - 1
        else:
            return True

    return False

File: minimum_common/min.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Solution
{
    public int GetCommon(int[] nums1, int[] nums2)
    {
        List<int> common = new List<int>();
        for (int i = 0; i < nums1.Length; i++)
        {
            if (binarySearch(nums2, nums1[i]) != -1)
            {
                common.Add(nums1[i]);
            }
        }

        return common.Count == 0 ? -1 : common.Min();
    }

    int binarySearch(int[] nums, int target)
    {
        int i = 0,
            j = nums.Length - 1;

        while (i <= j)
        {
            int mid = i + (j - i) / 2;

            if (nums[mid] > target)
            {
                j = mid - 1;
            }
            else if (nums[mid] < target)
            {
                i = mid + 1;
            }
            else
            {
                return target;
            }
        }

        return -1;
    }
}
